dutree/dutree.py

```diff
@@ -467,35 +481,31 @@ def human(value):
     return '{}   B'.format(value)
 
 
-def main():
-    pathname = None
-    if len(sys.argv) == 2:
-        def getsize(node):
-            return node.app_size()
-
-        pathname = sys.argv[1]
-        use_apparent_size = True
-
-    elif len(sys.argv) == 3:
-        def getsize(node):
-            return node.use_size()
+def main() -> None:
+    parser = argparse.ArgumentParser(prog='dutree', description='dutree shows a summary of the directories/files which take up the most space.')
+    parser.add_argument('--count-blocks', help='count block size not app size', action='store_true')
+    parser.add_argument('--xdev', help='stay on current fileystem device', action='store_true')
+    parser.add_argument('--no-skip-proc-sys', help='do not skip the filesystem devices for proc and sys directories, default these are skipped', action='store_true')
+    parser.add_argument('path')
 
-        if sys.argv[1] == '--count-blocks':
-            pathname = sys.argv[2]
-        elif sys.argv[2] == '--count-blocks':
-            pathname = sys.argv[1]
-        use_apparent_size = False
+    args = parser.parse_args()
 
-    if pathname is None:
-        sys.stderr.write('Usage: dutree [--count-blocks] PATH\n')
-        sys.exit(1)
+    use_apparent_size = True
+    def getsize(node):
+        return node.app_size()
 
-    run(pathname, use_apparent_size, getsize)
+    if args.count_blocks:
+            use_apparent_size = False
+            def getsize(node):
+                return node.use_size()
+    
+ 
+    run(args.path, use_apparent_size, getsize, args.xdev, args.no_skip_proc_sys)
 
 
-def run(pathname, use_apparent_size, getsize):
+def run(pathname: str, use_apparent_size: bool, getsize: Callable, xdev: bool, no_skip_proc_sys: bool) -> None:
     verbose = True and not use_apparent_size
-    scanner = DuScan(pathname)
+    scanner = DuScan(pathname, xdev, no_skip_proc_sys)
```

Comment:
Maybe it makes sense to initialize the allow/denylists here instead:
```
scanner = DuScan(pathname)
if xdev:
    scanner.skip_other_filesystems()
elif skip_proc_sys:
    scanner.skip_proc_sys_filesystems()
```
(See previous comment about inverting (no_)skip_proc_sys.)

To do the above, you'd have to disregard _some_ of the suggestions I made previously about `_make_dev_allowdeny()`.

Reply:
I have changed the setting up  of the allow deny lists as requested